### ISSUE TITLE
[BUG] Media Element Seek not firing

### ISSUE THREAD

**Issue by ne0rrmatrix** [CommunityToolkit/Maui#1012]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

Media Element fails to seek playback positions under conditions where windows or android works fine.
Examples include:
-if using mediaElement.MediaOpened it will not fire under any situation.
-if you are using StateChanged it will fire but only if you check its value before trying to seek.

I have no idea why the behavior in iOS is so different from Windows and Android? Is this intended to work this way?
I have include3d sample repo showing what works does not work. Notes in app with what you have to do to make iOS seek.

### Expected Behavior

Media element behavior should be the same across devices. Or please document differences for iOS.

### Steps To Reproduce

1. Open included solution and run it.
2. Or create new maui solution and add media element package.
3. Add an event to seek an arbitrary position at start of video
4. Set event to fire when opening video.

### Link to public reproduction project repository

https://github.com/ne0rrmatrix/MediaElementIOSBug

### Environment

```markdown
- .NET MAUI CommunityToolkit: 3.1.0
- OS: iOS
- .NET MAUI: 7.0
Packages used:

 <ItemGroup>
	  <PackageReference Include="CommunityToolkit.Maui" Version="1.0.0-preview3840" />
	  <PackageReference Include="CommunityToolkit.Maui.MediaElement" Version="1.0.0-preview3840" />
	  <PackageReference Include="CommunityToolkit.MVVM" Version="8.1.0" />
</ItemGroup>
```
```


### Anything else?

In code there is commented out code that demonstrates a work around. Is media element behavior intended to be different across platforms? And if so can it be documented?

**Comment by ne0rrmatrix** [comment]
Tested against latest build of Media Element from Main as of Feb 19th. Tried the build with ios fixes. Did not help.

**Comment by jfversluis** [comment]
> I have no idea why the behavior in iOS is so different from Windows and Android? Is this intended to work this way?

The whole concept of .NET MAUI is based on the fact that we map everything from an abstract layer (`MediaElement` in this case) to a platform implementation. So basically each platform is it's own implementation and can show it's own behavior and/or bugs. So seeing differences between platforms is definitely not uncommon.

Is this intended to work this way? No, we strive to implement all functionalities the same as much as possible so you, as the consuming developer, don't have to worry about the platform differences.

Thanks for reporting this! For a next time please report 1 potential bug per GitHub issue. That makes is a bit easier to discuss each of them. Thanks!

**Comment by jfversluis** [comment]
MediaOpened seems to work fine from the sample app that is currently on main. Same as the SeekTo functionality.

**Comment by jfversluis** [comment]
MediaOpened in your sample seems to work just fine

![image](https://user-images.githubusercontent.com/939291/220079383-d47b5cca-5dde-4874-a605-bdf5aff8b59f.png)

I'm guessing the `SeekTo` doesn't work because it comes to early, but this seems to be specific to iOS. I don't think this

```csharp
if (mediaElement.Position < TimeSpan.FromSeconds(200)) // seek does not fire without this line.
{
    mediaElement?.SeekTo(TimeSpan.FromSeconds(200));
}
```

Actually fixes it, it probably has more to do with the fact that the state has to be Playing before you can seek on iOS. Using this code it works, and I imagine this would work on other platforms as well:

```csharp
if (e.NewState == MediaElementState.Playing)
{
    mediaElement?.SeekTo(TimeSpan.FromSeconds(200));
}
```

**Comment by ne0rrmatrix** [comment]
TY for the time to replay. I have tried using:

```
public void IOS(object sender, MediaStateChangedEventArgs e)
    {
        Debug.WriteLine("Media opening IOS");
        if(e.NewState == MediaElementState.Playing)
        {
        mediaElement?.SeekTo(TimeSpan.FromSeconds(200));
        }
    }
```
The above code works perfectly for iOS.
But on windows it crashes to desktop with 

System.Runtime.InteropServices.COMException: 'The application called an interface that was marshalled for a different thread. (0x8001010E (RPC_E_WRONG_THREAD))'

So it is nice I have found a way to make this work on iOS now. But I will have to continue to use different code for windows and maybe android. Untested on Android atm with new code. I have tried many variations on windows. 

Currently this is the way i currently get windows to work with seeking. The key point is the calling has to be
```
mediaElement.MediaOpened += DoesNotWorkInIOS;
```

with function:

```
public void DoesNotWorkInIOS(object sender, EventArgs e)
    {
        Debug.WriteLine("Media Opened");
        mediaElement?.SeekTo(TimeSpan.FromSeconds(200)); //Never seeks in iOS when you use media opened. 
        Debug.WriteLine("Seeking 200 seconds"); 
    }
```

**Comment by jfversluis** [comment]
For Windows you might need to invoke it on the main thread, or I guess that is something we could add on our side.

**Comment by ne0rrmatrix** [comment]
I am honestly new enough to development and using dotnet that  I would not know how to differentiate threads when calling in media element. I would not even know how to go about that. I am just happy I have it working in my app now. TY for reply and help. It is super confusing when I spend a few days and not learn what I was hoping. It is all good,  I just hope I can figure out what is going on. I am happy with getting it working, but I don't understand what is going on under the hood. 

Understanding all of what is going on with media element and how it is implemented will be an ongoing goal for me so that I can avoid issues like this. But I think I need to do more reading and develop more skill. I hope to able to some time in the future contribute to this project and I need to continue learning how this works.

**Comment by jfversluis** [comment]
Change the code to

```csharp
MainThread.BeginInvokeOnMainThread(() =>
{
    if(e.NewState == MediaElementState.Playing)
    {
        mediaElement?.SeekTo(TimeSpan.FromSeconds(200));
    }
});
```

I think that should make it work for now, also on Windows. I've opened a PR to fix that on the library side so as of the next version you can remove the call to `MainThread.BeginInvokeOnMainThread`.

**Comment by ne0rrmatrix** [comment]
That fixes it! Now I just have to rewrite the handling logic for playback position records so that I can take advantage of that. I have to eithe rework my app to do it using main thread or just use what currently works. ATM what I posed before works for what I need. I am currently tracking playback of video files across a list. And what I posed earlier does that and works on all device types. But your code works. Unfortunately I will have to rewrite my code to take advantage of it. My code is dependant on current behavior. I will rewrite it later if behavior changes.
